lib/galaxy/tools/toolbox/base.py

```diff
@@ -704,6 +704,7 @@ def __watch_directory( self, directory, elems, integrated_elems, load_panel_dict
         def quick_load( tool_file, async=True ):
             try:
                 tool = self.load_tool( tool_file )
+                tool._lineage = self._lineage_map.register( tool )
```

Comment:
I already tried that in the initial lineage PR ... now if only I could remember why that didn't work :D